src/main/java/com/commercetools/sync/products/helpers/VariantReferenceResolver.java

```diff
@@ -162,19 +175,27 @@ static boolean isProductReference(@Nonnull final JsonNode referenceValue) {
             .orElse(false);
     }
 
+    private static boolean isCategoryReference(@Nonnull final JsonNode referenceValue) {
+        return getReferenceTypeIdIfReference(referenceValue)
+            .map(referenceTypeId -> Objects.equals(referenceTypeId, Category.referenceTypeId()))
+            .orElse(false);
+    }
+
     @Nonnull
     private static Optional<String> getReferenceTypeIdIfReference(@Nonnull final JsonNode referenceValue) {
```

Comment:
what is `IfReference` means in here 🤔

Reply:
It was meant If the passed `JsonNode referenceValue` is a reference object. Basically, if not, an empty optional will be returned.

But reading it again, I also think it's confusing. Will remove it. It should be implicitly inferred from returning an optional.

Reply:
930f332